src/azure-cli-core/azure/cli/core/tests/test_api_profiles.py

```diff
@@ -111,13 +119,41 @@ def test_supported_api_version_min_constraint(self):
         with mock.patch('azure.cli.core.profiles._shared.AZURE_API_PROFILES', test_profile):
             self.assertTrue(supported_api_version(cli, ResourceType.MGMT_STORAGE, min_api='2000-01-01'))
 
+    def test_supported_api_version_min_constraint_semver(self):
+        cli = DummyCli()
+        cli.cloud = Cloud('TestCloud', profile='2017-01-01-profile')
+        test_profile = {'2017-01-01-profile': {ResourceType.MGMT_STORAGE: '7.0'}}
```

Comment:
nit, only if it is simple, can you use keyvault as the testing resource? so we know who is the trouble maker?

Reply:
I could add the ResourceType KEYVAULT indeed, and not add it in the Profile yet. As long as there is no CLI code that list all RT variables and expect this to match Profile (which would be bad anyway)

Reply:
Let us do it later till the KV command module gets updated to support multi-api. I thought the test could use any test string for resource type, but looks like it is not the case...